core/common/hexutil.hpp

```diff
@@ -59,6 +64,33 @@ namespace kagome::common {
    */
   std::string hex_lower_0x(const uint8_t *data, size_t size) noexcept;
 
+  template <std::output_iterator<uint8_t> Iter>
+  outcome::result<void> unhex_to(std::string_view hex, Iter out) {
```

Comment:
`out` is `vector` or `array` or `span`
```suggestion
  template <typename T>
  outcome::result<void> unhex_to(std::string_view hex, T &out) {
    if constexpr (requires { out.resize(); }) {
      out.resize(hex.size() / 2);
    } else if (hex.size() != out.size() * 2) {
      return ERROR;
    }
```